test/ccpm_test.go

```diff
@@ -0,0 +1,152 @@
+		{args: "ccpm plugin version describe ver-123456", fixture: "ccpm/plugin-version-describe-missing-plugin.golden", exitCode: 1},
+		{args: "ccpm plugin version delete --plugin ccp-123456", fixture: "ccpm/plugin-version-delete-missing-version.golden", exitCode: 1},
+		{args: "ccpm plugin version delete ver-123456", fixture: "ccpm/plugin-version-delete-missing-plugin.golden", exitCode: 1},
+	}
+
+	for _, test := range tests {
+		test.login = "cloud"
+		s.runIntegrationTest(test)
+	}
+}
+
+func (s *CLITestSuite) TestCCPMErrorHandling() {
+	tests := []CLITest{
+		// Network/API error handling - use different environment IDs for error cases
+		{args: "ccpm plugin list --environment env-error", fixture: "ccpm/plugin-list-api-error.golden", exitCode: 1},
+		{args: "ccpm plugin create --name my-plugin --cloud aws --environment env-error", fixture: "ccpm/plugin-create-api-error.golden", exitCode: 1},
+		{args: "ccpm plugin describe ccp-123456 --environment env-error", fixture: "ccpm/plugin-describe-api-error.golden", exitCode: 1},
+		{args: "ccpm plugin update ccp-123456 --name updated-name --environment env-error", fixture: "ccpm/plugin-update-api-error.golden", exitCode: 1},
+		{args: "ccpm plugin delete ccp-123456 --environment env-error --force", fixture: "ccpm/plugin-delete-api-error.golden", exitCode: 1},
+
+		// Invalid argument handling
+		{args: "ccpm plugin describe", fixture: "ccpm/plugin-describe-missing-id.golden", exitCode: 1},
+		{args: "ccpm plugin describe ccp-123456 ccp-789012 --environment env-123456", fixture: "ccpm/plugin-describe-too-many-args.golden", exitCode: 1},
+		{args: "ccpm plugin update", fixture: "ccpm/plugin-update-missing-id.golden", exitCode: 1},
+		{args: "ccpm plugin update ccp-123456 ccp-789012 --environment env-123456", fixture: "ccpm/plugin-update-too-many-args.golden", exitCode: 1},
+		{args: "ccpm plugin delete", fixture: "ccpm/plugin-delete-missing-id.golden", exitCode: 1},
+		{args: "ccpm plugin delete ccp-123456 ccp-789012 --environment env-123456", fixture: "ccpm/plugin-delete-too-many-args.golden", exitCode: 1},
+
+		// Version command invalid argument handling
+		{args: "ccpm plugin version describe", fixture: "ccpm/plugin-version-describe-missing-id.golden", exitCode: 1},
+		{args: "ccpm plugin version describe ver-123456 ver-789012 --plugin ccp-123456 --environment env-123456", fixture: "ccpm/plugin-version-describe-too-many-args.golden", exitCode: 1},
+		{args: "ccpm plugin version delete", fixture: "ccpm/plugin-version-delete-missing-id.golden", exitCode: 1},
+		{args: "ccpm plugin version delete ver-123456 ver-789012 --plugin ccp-123456 --environment env-123456", fixture: "ccpm/plugin-version-delete-too-many-args.golden", exitCode: 1},
+	}
+
+	for _, test := range tests {
+		test.login = "cloud"
+		s.runIntegrationTest(test)
+	}
+}
+
+func (s *CLITestSuite) TestCCPMAuthentication() {
```

Comment:
This is probably not needed.

If the authentication setting is wrong, then the `TestHelp` tests will generate an on-prem version, and that acts like a test of this.